src/update-feeds/check-feeds.ts

```diff
@@ -0,0 +1,198 @@
+          dapiInfo.dataFeedValue.value,
+          dapiInfo.dataFeedValue.timestamp,
+          newBeaconSetValue,
+          newBeaconSetTimestamp,
+          dapiInfo.updateParameters.heartbeatInterval,
+          adjustedDeviationThresholdCoefficient
+        )
+      ) {
+        return null;
+      }
+
+      return {
+        dapiInfo,
+        UpdatableBeacons: zip(dapiInfo.decodedDataFeed.beacons, beaconsSignedData).map(([beacon, signedData]) => ({
+          signedData: signedData!,
+          beaconId: beacon!.beaconId,
+        })),
+      };
+    })
+    .filter((UpdatableDapi): UpdatableDapi is UpdatableDapi => UpdatableDapi !== null);
+
+interface OnChainValue {
+  beaconId: string;
+  onChainValue: { timestamp: ethers.BigNumber; value: ethers.BigNumber };
+}
+
+export const callAndParseMulticall = async (
+  batch: string[],
+  providerName: ProviderName,
+  chainId: ChainId
+): Promise<OnChainValue[]> => {
+  const { config } = getState();
+  const chain = config.chains[chainId]!;
+  const { providers, contracts } = chain;
+
+  const provider = new ethers.providers.StaticJsonRpcProvider(providers[providerName]);
+  const server = getApi3ServerV1(contracts.Api3ServerV1, provider);
+  const voidSigner = new ethers.VoidSigner(ethers.constants.AddressZero, provider);
+
+  const feedCalldata = batch.map((beaconId) => ({
+    beaconId,
+    calldata: server.interface.encodeFunctionData('dataFeeds', [beaconId]),
+  }));
+
+  if (feedCalldata.length === 0) {
+    return [];
+  }
+
+  const multicallResult = await go(
+    async () => server.connect(voidSigner).callStatic.tryMulticall(feedCalldata.map((feed) => feed.calldata)),
+    { retries: 1 }
+  );
+
+  if (!multicallResult.success) {
+    logger.warn(`The multicall attempt to read potentially Updatable feed values has failed.`, {
+      error: multicallResult.error,
+    });
+    throw multicallResult.error;
+  }
+
+  const { successes, returndata } = multicallResult.data;
+  if (!(successes.length === feedCalldata.length && returndata.length === feedCalldata.length)) {
+    throw new Error(`The number of returned records from the read multicall call does not match the number requested.`);
+  }
+
+  return feedCalldata
+    .map(({ beaconId }, idx) => {
+      if (successes[idx]) {
+        const [value, timestamp] = ethers.utils.defaultAbiCoder.decode(['int224', 'uint32'], returndata[idx]!);
+
+        return {
+          beaconId,
+          onChainValue: { timestamp: ethers.BigNumber.from(timestamp), value: ethers.BigNumber.from(value) },
+        };
+      }
+
+      return null;
+    })
+    .filter((onChainValue): onChainValue is OnChainValue => onChainValue !== null);
+};
+
+export const deeplyCheckDapis = (
+  batch: ReturnType<typeof shallowCheckFeeds>,
+  deviationThresholdCoefficient: DeviationThresholdCoefficient,
+  onChainValues: Awaited<ReturnType<typeof callAndParseMulticall>>
+): UpdatableDapi[] =>
+  batch
+    .map(({ dapiInfo, UpdatableBeacons }) => {
```

Comment:
might need to lowercase?
```suggestion
    .map(({ dapiInfo, updatableBeacons }) => {
```